### PR TITLE
Kotlin Support

### PR DESCRIPTION
**Pull request by pocmo**
Enabling Kotlin support and replaced DisplayToolbar class with Kotlin implementation (for testing tooling etc.)

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/browser/DisplayToolbar.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    constructor(context: Context) : super(context)
+
+    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
+
+    override fun onFinishInflate() {
+        super.onFinishInflate()
+
+        toolbarContent = findViewById<View>(R.id.toolbar_content)
```
Comment (mcomella) [inline app/src/main/java/org/mozilla/focus/browser/DisplayToolbar.kt]: nit: As I mentioned in the previous commit, with kotlin android extensions, these lines are no longer necessary

File: app/build.gradle [hunk 2]
```diff
@@ -163,6 +165,8 @@ dependencies {
 
     compile 'org.mozilla.telemetry:telemetry:1.1.1'
 
+    compile "org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlin_version"
```
[the remark this answers is not in the document]
Comment (pocmo) [reply app/build.gradle]: Nice! I'll file follow-up bugs for those. What are the test libraries used for? I've written junit tests in Kotlin in an other patch and didn't need additional libraries.

The Android extensions are looking interesting. I'll file a bug to look at APK size etc.

File: build.gradle [hunk 3]
```diff
@@ -3,6 +3,7 @@
 buildscript {
     ext.support_libraries_version = '26.0.2'
     ext.espresso_version = '3.0.1'
+    ext.kotlin_version = '1.1.4-3'
```
[the remark this answers is not in the document]
Comment (pocmo) [reply build.gradle]: Thanks! This patch is indeed a little bit older already. :)

File: app/src/main/java/org/mozilla/focus/browser/DisplayToolbar.kt [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    private lateinit var progressView: AnimatedProgressBar
+
+    private val collapsedProgressTranslationY: Float by lazy {
+        TypedValue.applyDimension(
+                TypedValue.COMPLEX_UNIT_DIP, 2f, resources.displayMetrics)
+    }
+
+    constructor(context: Context) : super(context)
+
+    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
```
[the remark this answers is not in the document]
Comment (pocmo) [reply app/src/main/java/org/mozilla/focus/browser/DisplayToolbar.kt]: Nice!

File: app/src/main/java/org/mozilla/focus/browser/DisplayToolbar.kt (hunk 1 above)
Comment (pocmo) [reply app/src/main/java/org/mozilla/focus/browser/DisplayToolbar.kt]: I'll look at the extensions in #1411 and will update this part in a follow-up patch.

**Comment by pocmo** [comment]
🚀

File: app/build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (mcomella) [reply app/build.gradle]: > What are the test libraries used for?

To be honest, I'm not sure! :P All Java testing infrastructure should *just work* so I assume they're some type of testing constructs that work specifically for Kotlin features (e.g. maybe to work around how classes are final by default, iirc).

[the remark this answers is not in the document]
Comment (mcomella) [reply app/build.gradle]: > What are the test libraries used for?

To be honest, I'm not sure! :P All Java testing infrastructure should *just work* so I assume they're some type of testing constructs that work specifically for Kotlin features (e.g. maybe to work around how classes are final by default, iirc).

**Pull request merged**